src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java

```diff
@@ -34,249 +34,203 @@
- * configuration such an import should be placed at the end of the import list.
- * </p>
- * The rule consists of:
- *
- * <p>
- * 1. STATIC group. This group sets the ordering of static imports.
- * </p>
- *
- * <p>
- * 2. SAME_PACKAGE(n) group. This group sets the ordering of the same package imports.
- * Imports are considered on SAME_PACKAGE group if <b>n</b> first domains in package name
- * and import name are identical.
- * </p>
- *
- * <pre>
- *{@code
- *package java.util.concurrent.locks;
- *
- *import java.io.File;
- *import java.util.*; //#1
- *import java.util.List; //#2
- *import java.util.StringTokenizer; //#3
- *import java.util.concurrent.*; //#4
- *import java.util.concurrent.AbstractExecutorService; //#5
- *import java.util.concurrent.locks.LockSupport; //#6
- *import java.util.regex.Pattern; //#7
- *import java.util.regex.Matcher; //#8
- *}
- * </pre>
- *
- * <p>
- * If we have SAME_PACKAGE(3) on configuration file,
- * imports #4-6 will be considered as a SAME_PACKAGE group (java.util.concurrent.*,
- * java.util.concurrent.AbstractExecutorService, java.util.concurrent.locks.LockSupport).
- * SAME_PACKAGE(2) will include #1-8. SAME_PACKAGE(4) will include only #6.
- * SAME_PACKAGE(5) will result in no imports assigned to SAME_PACKAGE group because
- * actual package java.util.concurrent.locks has only 4 domains.
- * </p>
- *
- * <p>
- * 3. THIRD_PARTY_PACKAGE group. This group sets ordering of third party imports.
- * Third party imports are all imports except STATIC,
- * SAME_PACKAGE(n), STANDARD_JAVA_PACKAGE and SPECIAL_IMPORTS.
- * </p>
- *
- * <p>
- * 4. STANDARD_JAVA_PACKAGE group. By default this group sets ordering of standard java/javax
- * imports.
- * </p>
- *
- * <p>
- * 5. SPECIAL_IMPORTS group. This group may contains some imports
- * that have particular meaning for the user.
- * </p>
- *
- * <p>
- * NOTE!
- * </p>
- * <p>
- * Use the separator '###' between rules.
+ * Checks that the groups of import declarations appear in the order specified by the user.
+ * If there is an import but its group is not specified in the configuration such an import should
+ * be placed at the end of the import list.
  * </p>
  * <p>
- * To set RegExps for THIRD_PARTY_PACKAGE and STANDARD_JAVA_PACKAGE groups use
- * thirdPartyPackageRegExp and standardPackageRegExp options.
+ * <a href="#CustomImportOrder_Examples">Examples section</a> contains examples that work with
+ * default formatter configurations of Eclipse, IntelliJ IDEA and NetBeans
  * </p>
- * <p>
- * Pretty often one import can match more than one group. For example, static import from standard
- * package or regular expressions are configured to allow one import match multiple groups.
- * In this case, group will be assigned according to priorities:
- * </p>
- * <ol>
+ * <ul>
  * <li>
- *    STATIC has top priority
+ * Property {@code customImportOrderRules} - List of order declaration customizing by user.
```

Comment:
@romani I mean the description of the properties.
With the realignment of the javadocs, I also updated the first word of the property's descriptions because we use the same description for the setter.

Reply:
> Please relook at the other properties.

done.

Reply:
@pbludov The description for this property didn't change, nor do I see changes for ones that start with `RegExp`.

> Properties should start with a singular verb.

Reply:
done